src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,76 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static Map<Class<?>, Object> instances;
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+
+        Class<?> implClazz = getImpl(interfaceClazz);
+        Object clazzImplementationObject = createNewInstance(implClazz);
+        if (!implClazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Annotation @Component is not present in the class: "
+                    + implClazz.getName());
+        }
+        Field[] declaredFields = implClazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                field.setAccessible(true);
+                try {
+                    field.set(clazzImplementationObject, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't set the field: \"" + field.getName()
+                            + "\" of the class: \"" + implClazz.getName() + "\"");
+                }
+
+            }
+        }
+        return clazzImplementationObject;
+    }
+
+    private Object createNewInstance(Class<?> implClazz) {
+        if (instances == null) {
+            instances = new HashMap<>();
```

Comment:
I think it's better to initialize instances Map at the time of declaration (`private final Map<Class<?>, Object> instances = new HashMap<>();`)